File: fizzbuzz.py
#!/usr/bin/python

import sys
import re

# check that a filename argument was provided, otherwise
if len(sys.argv) < 2:
  raise Exception("Filename must be first argument provided")
  
filename = sys.argv[1]
lines = []

# open file in read mode, assuming file  is in same directory as script
try:
  file = open(filename, 'r')

  # read Fibbonacci indexes from file into list
  lines = file.readlines()
  file.close()
except IOError:
  print "File '%s' was not found in current directory" % filename

# strip out newline characters
lines = [line.replace('\n', '') for line in lines]

# ignore empty lines
try:
  lines.remove("")
except:
  pass

for line in lines:
  numbers = line.split(" ")
  a = int(numbers[0])
  b = int(numbers[1])
  num = int(numbers[2])
  
  current = 1
  digits = []
  while current <= num:
    digits.append(current)
    current += 1
    
  #print digits
  
  for digit in digits:
    new_value = ""
    index = digits.index(digit)
    
    divisible_by_a = digit % a == 0
    divisible_by_b = digit % b == 0
    divisible_by_both = divisible_by_a and divisible_by_b
    
    if divisible_by_a:
      new_value = "F"
    if divisible_by_b:
      new_value = "B"
    if divisible_by_both:
      new_value = "FB"
    if new_value:
      digits[index] = new_value
      
  digits = [str(d) for d in digits]
  print " ".join(digits)
  
  

File: fizzbuzz.rb
#!/usr/bin/ruby

file_name = ARGV[0]

if file_name.nil?
  exit
end

file = File.new(file_name)

while (line = file.gets)
  digits = line.split(" ")
  divisor_a = digits[0].to_i
  divisor_b = digits[1].to_i
  upto_value = digits[2].to_i
  numbers = []
  1.upto(upto_value) {|x| numbers.push(x)}
  numbers.each{ |x|
    idx = numbers.index(x)
    
    if x % divisor_a == 0
      numbers[idx] = 'F'
    end
    
    if x % divisor_b == 0
      numbers[idx] = 'B'
    end
    
    if x % divisor_a == 0 && x % divisor_b == 0
      numbers[idx] = 'FB'
    end
  }
  
  puts numbers.join(" ")
end